### ISSUE TITLE
Support training and evaluation of adversarial patches

### ISSUE THREAD

**Issue by deprit** [twosixlabs/armory-library#49]
Armory-library currently supports PGD-style attacks where each image is adversarially perturbed. Another style of evasion attack produces a single adversarial patch on a training set of images. The patch is then applied to a test set, and the accuracy of the classifier or detector measured.

Armory-Library should support the training of and evaluation of adversarial patches as illustrated in this [ART notebook](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/notebooks/adversarial_patch/attack_adversarial_patch.ipynb).
